### PR TITLE
Add usages for the constraints.

### PR DESCRIPTION
**Pull request by tamdaz**
## Context

Reference to issue #201

This pull request consists to add usages that provide a better understanding of their representations, so, developers can easily use the constraints.

## Changelog

- Add example usages to `AVD::Constraints` docs

---

_Before merging, remember to add the `athena-framework/athena` prefix to the PR number in the PR title_

### CODE REVIEW

File: src/components/validator/src/constraints/optional.cr
Comment (Blacksmoke16) [inline src/components/validator/src/constraints/optional.cr:1]: `Optional` is a bit special so think we should skip adding an example for this one as it's not entirely correct.

File: src/components/validator/src/constraints/required.cr
Comment (Blacksmoke16) [inline src/components/validator/src/constraints/required.cr:1]: `Required` is also a bit special so think we should skip adding an example for this one as it's not entirely correct.

**Review by Blacksmoke16 — CHANGES_REQUESTED** [review]
Thanks for this. Overall it looks good to me, but left a couple comments.

File: src/components/validator/src/constraints/optional.cr
Comment (tamdaz) [reply src/components/validator/src/constraints/optional.cr:1]: resolved

File: src/components/validator/src/constraints/required.cr
Comment (tamdaz) [reply src/components/validator/src/constraints/required.cr:1]: resolved

**Review by Blacksmoke16 — APPROVED** [review]

**Comment by tamdaz** [comment]
Perfect, thank you. :)

**Pull request merged**